tests/snapshot.test.ts

```diff
@@ -69,6 +74,41 @@ it('should create a new proxy from a snapshot', async () => {
   expect(state2.c).toBe(0)
 })
 
+it('should return updated property snapshot when property is updated after removed from Array and added to array again', async () => {
+  /**
+   * error occurs when
+   * [1]. Remove target item should be in the array
+   * [2]. Remove target item should be nested object
+   * [3]. The array should be subscribed
+   * [4]. The array should pass to snapshot() before the target item removed
+   */
```

Comment:
@dai-shi I added test and reverted the previous commit.
And wrote the error occurrence conditions.

Reply:
Thanks. Can you resolve conflicts and merge the main branch?

Reply:
Done! And I will look in to it about another approach that you suggested. 😀

Reply:
Hi, are you still around?

Reply:
I made some tryies after this but it didn't worked. (ex- force version update if child object removed)
I think It will be hard to fix without breaking something as you said.

Do you have any idea about approach to fixit?

Reply:
https://github.com/pmndrs/valtio/issues/712#issuecomment-1643355452 is my current understanding.